storage/Devices/MdImpl.cc

```diff
@@ -383,8 +393,13 @@ namespace storage
 	    try
 	    {
 		const MdadmDetail& mdadm_detail = system_info.getMdadmDetail(name);
+
 		if (!mdadm_detail.devname.empty())
-		    name = DEV_MD_DIR "/" + mdadm_detail.devname;
+                {
+		    string devname = DEV_MD_DIR "/" + mdadm_detail.devname;
+                    if (file_exists(devname))
```

Comment:
During probing everything has to use SystemInfo for mocking in the testsuite and remote probing.